internal/controllers/synthesis/pod.go

```diff
@@ -25,42 +27,74 @@ func newPod(cfg *Config, comp *apiv1.Composition, syn *apiv1.Synthesizer) *corev
 		"eno.azure.io/synthesizer-generation": strconv.FormatInt(syn.Generation, 10),
 	}
 
-	userID := int64(1000)
-	yes := true
 	pod.Spec = corev1.PodSpec{
 		ServiceAccountName: cfg.PodServiceAccount,
+		Affinity: &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{{
+					Weight: 100,
+					PodAffinityTerm: corev1.PodAffinityTerm{
+						TopologyKey: "kubernetes.io/hostname",
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								manager.ManagerLabelKey: manager.ManagerLabelValue,
+							},
+						},
+					},
+				}},
+			},
+		},
 		Containers: []corev1.Container{{
 			Name:    "synthesizer",
 			Image:   syn.Spec.Image,
 			Command: []string{"sleep", "infinity"},
 			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: ptr.To(false),
+				ReadOnlyRootFilesystem:   ptr.To(true),
+				RunAsUser:                ptr.To(int64(65532)),
```

Comment:
This is not guaranteed to be correct, but I _think_ that it's not a big deal: worst case scenario, the Pod doesn't use distroless or equivalent as a base image which _hopefully_ leads to a terminal error. In any case it's only a problem for OSS 😜